### PR TITLE
Add schema-related test cases to `tjson` package

### LINKED ISSUE #943: `tjson`: Add negative schema-related test cases
This task is a part of #683 epic.

Currently, in `tjson` tests we cover the simplest cases of dealing with schemas (valid schemas that correspond to the given types).
Let's add some invalid schemas and fuzzing. It does not have to be very effective.

To Do:
- [x] Test data types with valid schemas that don't represent the given type
- [x] Test data types with invalid schemas
- [x] Add schema fuzzing to the `fuzzJSON` function with something like:
```go
f.Add(tc.canonJ, schemaAsString)

// ...

f.Fuzz(func(t *testing.T, j, schema string) {
```

### PR DESCRIPTION
**Pull request by w84thesun**
# Description

This PR closes #943.

## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [ ] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: internal/tjson/tjson_test.go [hunk 1]
```diff
@@ -222,28 +229,38 @@ func testJSON(t *testing.T, testCases []testCase, newFunc func() tjsontype) {
...
 		// raw "null" should never reach UnmarshalJSON due to the way encoding/json works
 		if j == "null" {
 			t.Skip()
 		}
 
+		s := newSchema()
+
+		err := s.Unmarshal([]byte(schema))
+		if err != nil {
+			t.Skip()
```
Comment (chilagrow) [inline internal/tjson/tjson_test.go:256]: This skip is for invalid schema only right?

**Review by chilagrow — APPROVED** [review]
Looks good to me :hugs:

File: internal/tjson/tjson_test.go [hunk 2]
```diff
@@ -222,28 +229,38 @@ func testJSON(t *testing.T, testCases []testCase, newFunc func() tjsontype) {
 
 func fuzzJSON(f *testing.F, testCases []testCase) {
 	for _, tc := range testCases {
-		f.Add(tc.j)
+		schema, err := tc.schema.Marshal()
+		if err != nil {
+			f.Fatalf("failed to marshal schema: %v", err)
```
Comment (AlekSi) [inline internal/tjson/tjson_test.go:234]: Why not `require.NoError(f, err)` as we do usually?

File: internal/tjson/tjson_test.go (hunk 1 above)
Comment (AlekSi) [reply internal/tjson/tjson_test.go:256]: Seems so

File: internal/tjson/tjson_test.go (hunk 2 above)
Comment (w84thesun) [reply internal/tjson/tjson_test.go]: No reason, it is left after debugging.

File: internal/tjson/tjson_test.go [hunk 3]
```diff
@@ -231,7 +231,7 @@ func fuzzJSON(f *testing.F, testCases []testCase) {
 	for _, tc := range testCases {
 		schema, err := tc.schema.Marshal()
 		if err != nil {
-			f.Fatalf("failed to marshal schema: %v", err)
+			require.NoError(f, err)
```
Comment (AlekSi) [inline internal/tjson/tjson_test.go:234]: c'mon, why need `if err != nil` then?

**Review by AlekSi — APPROVED** [review]

**Review by rumyantseva — COMMENTED** [review]
As discussed at daily sync, the changes look good to me.

One question - do you understand why `Go / Fuzz short` fails? Is it expected?

**Comment by w84thesun** [comment]
> As discussed at daily sync, the changes look good to me.
> 
> One question - do you understand why `Go / Fuzz short` fails? Is it expected?

Now it's not failing. It seems that it was because of #1274.

**Review by rumyantseva — APPROVED** [review]
LGTM

**Pull request merged**